ArrayOrder.md

```diff
@@ -0,0 +1,202 @@
+## Multi-dimensional array indexing
+
+Zarr stores multi-dimensional arrays into regularly sized chunks.
+Chunks are themselves multi-dimensional arrays of a smaller size than
+the complete multidimensional array and are stored as a 1D array of
+values, called a "flattened" array.
+
+Indexing into a multi-dimensional array is done with an ordered tuple,
+each element of which indexes into one of the array's dimensions.
+Elements of this tuple are called "coordinates." For example, the tuple
+`(i,j,k)` indexes a three-dimensional array, and `i`, `j`, and `k` are
+its coordinates. We will call `i` the "left" or "first" index, and `k` the
+"right" or "last" index.
+
+The only valid coordinates for arrays are the non-negative integers.
+
+The following will describe conventions for mapping multi-dimensional
+indexes (ordered tuples of integers) to the scalar integer index of the
+flattened 1D array.
+
+### Reshaping and stride
+
+One can think of reshaping a 1D array as a recursive process of grouping a
+number of adjacent elements.
+
+An (n-1)-dimensional array is reshaped to an n-dimensional array by
+grouping a number adjacent elements belonging to the same dimension. 
+
+* **Define:** the "size" of a dimension is the number of grouped elements.
+
+* **Define:** the stride of a dimension is the (positive) step in the
+  flat array that corresponds to the adjacent element along that dimension.
+
+The stride of a dimension is the product of sizes of all previous dimensions. 
```

Comment:
What does "previous" mean here? It seems that there is an implicit notion of "direction" baked in to a tuple representing dimensions that can either be "first to last" or "last to first". Can we make this concept more explicit?

Reply:
Agreed.  "all previous" means something like all dimensions of smaller stride, which as you say, is confusingly not related to whether the tuple is "first to last" or vice versa.